### DISCUSSION TITLE
Upgrade / re-Install behavior is different for casks and non-casks

### DISCUSSION THREAD

**Discussion by ee-usgs** [Homebrew/discussions#3048] [Casks]
I'm trying to create a script I can run to set up a new machine and occasionally re-run to update all my software.  The problem is that different formula behave differently on update - some just update, others tell me to reinstall the formula.

For instance, if I have this in my script:
```
brew install git

git 2.34.1 is already installed but outdated (so it will be upgraded).    <<-- Great!
```

But if I have this:
```
brew install dbeaver-community

Warning: Cask 'dbeaver-community' is already installed.

To re-install dbeaver-community, run:
  brew reinstall --cask dbeaver-community      <<--- No!  Just do an update!
```

Can't these just behave the same?  Or is there a command option to make them install / upgrade / reinstall as needed?

Thank you,
--ee

**Comment by SMillerDev** [comment]
Why not use `brew bundle` that manage the installs?

**Comment by cho-m** [comment]
Casks like `dbeaver-community` enable `auto_updates true` since they offer an internal auto-updater, which can lead to unwanted behavior when simultaneously updating via Homebrew https://docs.brew.sh/FAQ#why-arent-some-apps-included-during-brew-upgrade

For `brew upgrade`, there are some args (`--greedy`, `--greedy-auto-updates`) to bypass this described at https://docs.brew.sh/Manpage#upgrade-options-outdated_formulaoutdated_cask-

The recommended Brewfile can be useful to enable this for a subset of Casks https://github.com/Homebrew/homebrew-bundle#usage
